libs/vkd3d/state.c

```diff
@@ -3959,6 +3970,12 @@ uint32_t d3d12_graphics_pipeline_state_get_dynamic_state_flags(struct d3d12_pipe
     if (graphics->index_buffer_strip_cut_value && !is_mesh_pipeline)
         dynamic_state_flags |= VKD3D_DYNAMIC_STATE_PRIMITIVE_RESTART;
 
+    /* Enable dynamic sample count for multisampled pipelines so we can work around
+     * bugs where the app may render to a single sampled render target. */
+    if ((!key || !key->rasterization_samples) && d3d12_graphics_pipeline_needs_dynamic_rasterization_samples(&state->graphics) &&
```

Comment:
This seems somewhat strange. I would expect `!key || key->rasterization_samples`? If rasterization_samples is 0, i.e. no RTs, rasterization samples is static.

Reply:
Removed this check. It was entirely redundant, `key->rasterization_samples` is 0 iff EDS3 rasterization samples is supported, see `get_or_create_pipeline` - the reasoning here is that we'd otherwise create redundant pipelines if we hit the fallback path for any other reason (DSV format memes without GPL).

There was a related issue though, `get_pipeline` wouldn't check for `dyn_state->rasterization_samples == 0` and instead we'd create a redundant fallback pipeline with the same properties as the main one. Fixed that too.